supertokens/supertokens.go

```
@@ -0,0 +1,161 @@
+package supertokens
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strings"
+)
+
+type SuperTokens struct {
+	instance          *SuperTokens
+	AppInfo           NormalisedAppinfo
+	IsInServerlessEnv bool
+	RecipeModules     []RecipeModule
+}
+
+func NewSuperTokens(config TypeInput) (*SuperTokens, error) {
+	var err error
+	var s *SuperTokens
+	s.AppInfo, err = NormaliseInputAppInfoOrThrowError(config.AppInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	hostList := strings.Split(config.Supertoken.ConnectionURI, ";")
+	var hosts []NormalisedURLDomain
+	for _, h := range hostList {
+		host, err := NewNormalisedURLDomain(h, false)
+		if err != nil {
+			fmt.Println(err)
+		}
+		hosts = append(hosts, *host)
+	}
+
+	q := Querier{}
+	q.Init(hosts, config.Supertoken.APIKey)
+
+	if config.RecipeList == nil || len(config.RecipeList) == 0 {
+		return nil, errors.New("Please provide at least one recipe to the supertokens.init function call")
+	}
+
+	s.IsInServerlessEnv = config.IsInServerlessEnv
+
+	for _, elem := range config.RecipeList {
+		s.RecipeModules = append(s.RecipeModules, elem(s.AppInfo, s.IsInServerlessEnv))
+	}
+
+	telemetry := config.Telemetry
+
+	if telemetry {
+		err := s.SendTelemetry()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return s, nil
+}
+
+func (s *SuperTokens) Init(config TypeInput) (err error) {
+	if s.instance == nil {
+		s.instance, err = NewSuperTokens(config)
+		return err
+	}
+	return nil
+}
+
+func (s *SuperTokens) GetInstanceOrThrowError() (*SuperTokens, error) {
+	if s.instance != nil {
+		return s.instance, nil
+	}
+	return nil, errors.New("Initialisation not done. Did you forget to call the SuperTokens.init function?")
+}
+
+func (s *SuperTokens) SendTelemetry() error {
```

We need to ignore any errors generated in this `SendTelemetry` function.